server/lib/pbench/indexer.py

```diff
@@ -3483,22 +3523,38 @@ def valid_ip(address):
 
 def search_by_host(sos_d_list, host):
     for sos_d in sos_d_list:
-        if host in sos_d.values():
-            return sos_d["hostname-f"]
+        try:
+            host_f = sos_d["hostname-f"]
+        except KeyError:
+            continue
+        else:
+            if host == host_f:
+                return host_f
+        try:
+            host_s = sos_d["hostname-s"]
+        except KeyError:
+            continue
+        else:
+            if host == host_s:
+                return host_f
     return None
 
 
 def search_by_ip(sos_d_list, ip):
     # import pdb; pdb.set_trace()
     for sos_d in sos_d_list:
+        try:
+            host_f = sos_d["hostname-f"]
+        except KeyError:
+            continue
         for l in sos_d.values():
-            if isinstance(l, list):
+            if not isinstance(l, list):
                 continue
             for d in l:
-                if isinstance(d, dict):
+                if not isinstance(d, dict):
```

Comment:
How many bugs can one pack in 4 lines of code?

Reply:
A lot when you make a bad code change (PR #1536) and we don't have a unit test to catch the bug.

Reply:
I thought I had introduced those bugs from day 1 and I was being self-deprecating - next time, I'll check before opening my mouth :-)